src/exporters/vhi.py

```diff
@@ -0,0 +1,220 @@
+    if output_filename.exists():
+        print(f"File already exists! {output_filename}")
+        return
+
+    # download to output_filename
+    with open(output_filename, 'wb') as out_f:
+        ftp_instance.retrbinary("RETR " + filename, out_f.write)
+
+    if output_filename.exists():
+        print(f"Successful Download! {output_filename}")
+    else:
+        print(f"Error Downloading file: {output_filename}")
+
+    return
+
+
+def batch_ftp_request(args: Dict, filenames: List) -> None:
+    # unpack multiple arguments
+    raw_folder = args['raw_folder']
+    years = args['years']
+
+    legit_filenames = [f for f in filenames if str(years) in f]
+    # create one FTP connection for each batch
+    with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+        ftp.login()
+        ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+        # download each filename using this FTP object
+        for raw_filename in legit_filenames:
+            output_filename = (
+                make_filename(raw_folder, raw_filename, dataset='vhi')
+            )
+            download_file_from_ftp(ftp, raw_filename, output_filename)
+
+    return
+
+
+class VHIExporter(BaseExporter):
+    """Exports Vegetation Health Index from NASA site
+
+    ftp.star.nesdis.noaa.gov
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+    @staticmethod
+    def get_ftp_filenames() -> List:
+        """  get the filenames containing VHI """
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # append the filenames to a list
+            listing: List = []
+            ftp.retrlines("LIST", listing.append)
+            # extract the filename
+            filepaths = [f.split(' ')[-1] for f in listing]
+            # extract only the filenames of interest
+            vhi_files = [f for f in filepaths if ".VH.nc" in f]
+
+        return vhi_files
+
+    # @staticmethod
+    def chunks(self, l: List, n: int) -> Generator:
+        # return a generator object which chunks list into sublists of size n
+        # https://chrisalbon.com/python/data_wrangling/break_list_into_chunks_of_equal_size/
+        # For item i in a range that is a length of l,
+        for i in range(0, len(l), n):
+            # Create an index range for l of n items:
+            yield l[i:i + n]
+
+    def save_errors(self, outputs: List) -> None:
+        print("\nError: ", [errors for errors in outputs if errors is not None])
+
+        # save the filenames that failed to a pickle object
+        with open(self.raw_folder / 'vhi_export_errors.pkl', 'wb') as f:
+            pickle.dump([error[-1] for error in outputs if error is not None], f)
+
+        return
+
+    def run_parallel(self,
+                     vhi_files: List,
+                     years: List[int] = [yr for yr in np.arange(1981, 2020)]
```

Comment:
We shouldn't have mutable values as default arguments. One option is to have a function which creates default arguments, like the ERA5 exporter's [`get_era5_times`](https://github.com/esowc/ml_drought/blob/master/src/exporters/cds.py#L129)

Reply:
hmm not sure i've understood this - can you check if this fixed in the new push?